app/src/main/java/com/example/atheneum/fragments/AddBookFragment.java

```diff
@@ -27,11 +54,129 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         this.view = inflater.inflate(R.layout.fragment_add_edit_book, container, false);
 
         this.context = getContext();
+
+        // bind editText references to UI elements
+        titleEditText = this.view.findViewById(R.id.bookTitleEditText);
+        authorEditText = this.view.findViewById(R.id.authorEditText);
+        isbnEditText = this.view.findViewById(R.id.isbnEditText);
+        descEditText = this.view.findViewById(R.id.descEditText);
+
+        saveBtn = this.view.findViewById(R.id.saveBookBtn);
+        saveBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                saveNewBook();
+            }
+        });
+        scanIsbnBtn = this.view.findViewById(R.id.scanIsbnButton);
+        scanIsbnBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                scanIsbn();
+            }
+        });
+        autoPopulateByIsgnBtn = this.view.findViewById(R.id.populateFromIsbnBtn);
+        autoPopulateByIsgnBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                populateFieldsByIsbn();
+            }
+        });
+
         mainActivity = (MainActivity) getActivity();
         // set action bar title
         mainActivity.setActionBarTitle(context.getResources().getString(R.string.add_book_prompt));
 
         return this.view;
     }
 
+    public void scanIsbn() {
+        Log.i("AddBook", "AddBook*** ISBN scan requested");
+        // TODO
+    }
+
+    public void populateFieldsByIsbn() {
+        // TODO: FINISH
+
+        Log.i("AddBook", "AddBook*** Auto-populate requested");
+        long isbn = -1;
+        if (!isbnEditText.getText().toString().equals("")) {
+             isbn = Integer.parseInt(isbnEditText.getText().toString());
+
+        }
+        else {
+            Log.i("AddBook", "AddBook*** No ISBN ");
+            // TODO: Print error message
+
+        }
+
+    }
+
+    public boolean allFieldsFilled() {
```

Comment:
It is a bit more work but have you thought about using a TextWatcher to validate each individual edit text [as the user enters in information](https://stackoverflow.com/a/11838715). That way you could validate text as the user enters input and if you made an array of these validators connected to their edit texts, then you could call validate() on all the edittexts at once instead of doing this big and statement. I could show you how I used this for Assignment 0 if you need a more concrete idea.

But this isn't strictly necessary and is just a nice UI touch.

Reply:
Yeah that would be good. I'll add it in a seperate merge request later; if you could show me how you did it for the assignment it would be a big help.